modules/lang-painless/spi/src/main/java/org/elasticsearch/painless/spi/annotation/AliasAnnotation.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.painless.spi.annotation;
+
+/**
+ * Creates an alias in PainlessLookupBuilder for the given class.  Can be used to expose an inner class without
+ * scripts and whitelists needing to scope it by the outer class.
+ *
+ * For class
+ *
+ * <pre>
+ * public class Outer {
+ *     public static class Inner {
+ *
+ *     }
+ *     public Inner inner() {
+ *         return new Inner();
+ *     }
+ * }
+ * </pre>
+ *
+ * Normally scripts would need to reference <pre>Outer.Inner</pre>.
+ *
+ * With an alias annotation <pre>@alias[class="Inner"]</pre> on the class
+ * <pre>
+ * class Outer$Inner @alias[class="AliasedTestInnerClass"] {
+ * }
+ * </pre>
+ *
+ * Then whitelist can have <pre>Inner</pre> as the return value for <pre>inner</pre> instead of <pre>Outer.Inner</pre>
+ * <pre>
+ * class Outer {
+ *   Inner inner()
+ * }
+ * </pre>
+ *
+ * And scripts refer can to <pre>Inner</pre> directly, <pre>Inner inner = Outer.inner()</pre>, instead of using the outer class to scope
+ * the type name <pre>Outer.Inner</pre> as would normally be required <pre>Outer.Inner inner = Outer.inner()</pre>
```

Comment:
<pre> is for blocks, I think you want `{@code ...}` here.

Reply:
Coded.